test/net/sourceforge/kolmafia/textui/command/HeistCommandTest.java

```diff
@@ -0,0 +1,116 @@
+package net.sourceforge.kolmafia.textui.command;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import net.sourceforge.kolmafia.FamiliarData;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.objectpool.FamiliarPool;
+import net.sourceforge.kolmafia.request.GenericRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class HeistCommandTest extends AbstractCommandTestBase {
+  @BeforeEach
+  public void initEach() {
+    KoLCharacter.reset("testUser");
+    new HeistCommandFakeRequest().register("heistFake");
```

Comment:
This is one way of doing this for sure! I think there's a way of sneaking in to GenericRequest and telling it what to return, but I think to have easy access to those you need to be extending RequestBase

Reply:
yeah, I'm already extending `AbstractCommandTestBase` here so I just went for a factory method pattern. Works well. Before DI, this was one popular way of testing classes with dependencies (in languages that let you extend classes and override methods).